File: C in Vscode/Search Algorithms/phonebook.js
// Linear Search
//
const person = [
    {
        name: "Carter",
        number: "+8801999577330"
    },
    {
        name: "David",
        number: "+69"
    }
];

const findNumber = () => {
    for (let i = 0; i < 2; i++) {
        console.log("Attempt", i);

        if (person[i].name == "Carter") {
            console.log("Found",person[i].name, person[i].number);
            return;
        }

    };
    console.log("Not found");
};

findNumber();

File: C in Vscode/Search Algorithms/phonebook.c
#include <stdio.h>
#include <string.h>

typedef struct
{
    char name[6];
    char number[50];
}
person;



int main(void)
{
    person people[2];
    people[0].name="Carter";


    const char *names[] = {"Bill", "Charlie"};
    const char *numbers[] = {"+8801999577330", "+8801999577330"};

    for (int i = 0; i < 2; i++)
    {
        printf("Attempt %i\n", i);
        if (names[i] = "Bill")
        {
            printf("Found\n");
            printf("%c", names[i]);
            return 0;
        }
    }
    printf("Not Found\n");
    return 1;
}